### ISSUE TITLE
[Portal Loop] Docker support for local dev environment

### ISSUE THREAD

**Issue by zivkovicmilos** [gnolang/gno#1204]
## Description

This task outlines the required Docker support for #1131 to have good UX. Docker will be used in combination with a `Makefile` to make the local development environment as seamless as possible for the user.

The Docker functionality that is required:
- The latest `master` or specific commit hash is cloned, and relevant `gno` binaries are built out
- Upon node start, transactions are replayed using the tx-archive tool (pulled from Docker Hub)
- Upon node stop, before the node is shut down, transactions are exported using the tx-archive tool (pulled from Docker Hub)
- An instance of the [gnolang/faucet](https://github.com/gnolang/faucet) is started in parallel with the node (`test1` can be used as the faucet address)
- An instance of [gnoweb](https://github.com/gnolang/gno/tree/master/gno.land/cmd/gnoweb) is started in parallel with the node (no captcha requirement for the faucet)

**Note**: When #1203 is done and ready, the Docker functionality should incorporate it for (separate PR and issue):
- initial balance management
- initial transaction management

Working branch should be:
[feat/local-feedback-loop](https://github.com/gnolang/gno/tree/feat/local-feedback-loop)

**Comment by thehowl** [comment]
https://github.com/gnolang/gno/pull/1238#issuecomment-2766456114
